src/core/ddsi/src/ddsi_plist.c

```diff
@@ -3248,11 +3249,13 @@ const dds_qos_t ddsi_default_qos_reader = {
   .type_consistency.ignore_member_names = false,
   .type_consistency.prevent_type_widening = false,
   .type_consistency.force_type_validation = false,
-  .ignore_locator_type = 0
+  .ignore_locator_type = 0,
+  .data_representation.value.n = 0,
```

Comment:
I think it's not the encoding version which depends on the type, but the encoding format (plain, delimited, parametrized). To be compatible with non-XTypes nodes, we'll have to use the PLAIN_CDR (version 1) format for endpoints using a final type, so that the data representation QoS of the XTypes enabled node will match a non-XTypes node that has not set this QoS. 

For appendable and mutable data types, we'll support XCDR2 encoding (and not XCDR1), so this will be using the DELIMITED_CDR and PL_CDR2 formats (see also table 35 from the XTypes spec). Using an appendable or mutable type for an endpoint without explicitly setting the data representation QoS value to XCDR2 (and thus it defaults to XCDR1) will result in an `unsupported` error.